internal/app/oauth/storage.go

```diff
@@ -0,0 +1,337 @@
+package oauth
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"math/big"
+	mRand "math/rand"
+	"net"
+	"time"
+
+	"github.com/caos/oidc/pkg/oidc"
+	"github.com/caos/oidc/pkg/op"
+	"github.com/oklog/ulid"
+	"gopkg.in/square/go-jose.v2"
+)
+
+var (
+	appConfig *AuthSrvConfig
+	clients   map[string]op.Client
+	requests  map[string]AuthRequest
+	keySet    jose.JSONWebKeySet
+)
+
+type AuthStorage struct {
+	key *rsa.PrivateKey
+}
+
+type AuthSrvConfig struct {
+	Domain string `default:"http://localhost:8080/"`
+	Debug  bool
+}
+
+func NewAuthStorage(cfg *AuthSrvConfig) op.Storage {
+
+	appConfig = cfg
+
+	s := &AuthStorage{}
+
+	initData()
+	initKeys(s)
+
+	return s
+}
+
+func initData() {
+	clients = map[string]op.Client{
+		"01FH69GFQ4DFCXS5XD91JK4HZ1": &ConfClient{
+			ID:              "01FH69GFQ4DFCXS5XD91JK4HZ1",
+			applicationType: op.ApplicationTypeNative,
+			authMethod:      oidc.AuthMethodNone,
+			accessTokenType: op.AccessTokenTypeJWT,
+			responseTypes:   []oidc.ResponseType{oidc.ResponseTypeCode},
+			grantTypes:      []oidc.GrantType{oidc.GrantTypeCode, oidc.GrantTypeRefreshToken},
+			redirectURIs:    []string{"http://localhost:3000"},
+			allowedScopes:   []string{"openid", "profile", "email"},
+			loginURI:        "http://localhost:3000/login?id=%s",
+			iDTokenLifetime: 5 * time.Minute,
+			clockSkew:       0,
+			devMode:         true,
+		},
+	}
```

Comment:
```suggestion
	client := initClient(dev)
	clients = map[string]op.Client{
	   client.GetID(): client,
	}
```

Separate this go file to define the shared client.

```go
func initClient(dev bool) op.Client {
	return &ConfClient{
		// ...
		devMode: dev,
	}
}
```

Also consider:

- `applicationType `: Have you considered `ApplicationTypeWeb`?
- `authMethod`: Make sure that `AuthMethodNone` is ok. I think it means the client has no client secret.

Reply:
The `applicationType` changed into `ApplicationTypeWeb`. 
For the `authMethod` I think its ok to be `AuthMethodNone` as we can consider our front end app as a public client.